### ISSUE TITLE
what's the corresponding char* type in go

### ISSUE THREAD

**Issue by TeavenX** [ebitengine/purego#150]
### base info
platform: darwin/arm64
CGO=1

### question
hello, I have c func like this
```c
ABCXXX const char* Do_Something(void* a, int b);
ABCXXX char* Process_pre_func_output(const char* row_data, int row_data_len);
```
I wonder what's the corresponding `char*` and `const char*` type in go
I've tried `*string` `*[]byte` `*[]rune` `unsafe.Pointer`, however none of them worked

ps: not work means, c func received an invalid argument OR can not set correct value to return type OR wrong mem address error

btw, it's an awesome project!🫡

**Comment by TeavenX** [comment]
found a tortuous solution, takes me hours, :(
```go
var doSomething func(a unsafe.Pointer, b int) unsafe.Pointer
var processPreFuncOutput func(row_data unsafe.Pointer, row_data_len int) unsafe.Pointer

row, rowLen := doSomething(a, b)
tmp := processPreFuncOutput(row, rowLen)

res := (*string)(unsafe.Pointer(&reflect.StringHeader{Data: uintptr(tmp), Len: tmp_len}))
```
well, it's even better if there is a type mapping

**Comment by TotallyGamerJet** [comment]
char* is a *byte bc both are a byte big. If you know the length u can convert that pointer into a slice using unsafe.Slice.

**Comment by hajimehoshi** [comment]
https://github.com/ebitengine/purego/commit/e5b87e976b386838bf71ee85f2eb03f753bae8b9

As this is already documented with an example, there is no further action item for this issue. What do you think? @TotallyGamerJet

**Comment by TotallyGamerJet** [comment]
Agreed. The PR should probably also be closed since it doesn't run.

**Comment by hajimehoshi** [comment]
Closed per e5b87e976b386838bf71ee85f2eb03f753bae8b9
